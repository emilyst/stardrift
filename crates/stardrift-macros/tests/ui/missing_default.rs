// This test should fail to compile because field2 is missing a default attribute
use stardrift_macros::ConfigDefaults;

#[derive(ConfigDefaults)]
struct MissingDefault {
    #[default(42)]
    pub field1: i32,

    // Missing #[default(...)]
    pub field2: String,
}

fn main() {}
