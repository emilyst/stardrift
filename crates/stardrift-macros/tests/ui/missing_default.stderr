error: Field 'field2' must have a #[default(...)] attribute specifying its default value
  --> tests/ui/missing_default.rs:10:5
   |
10 |     pub field2: String,
   |     ^^^^^^^^^^^^^^^^^^
